src/txn/txn_rollback_to_stable.c

```diff
@@ -424,7 +424,7 @@ __rollback_abort_row_ondisk_kv(
         } else
             return (0);
     } else if (vpack->tw.durable_start_ts > rollback_timestamp ||
-      (vpack->tw.durable_stop_ts == WT_TS_NONE && prepared)) {
+      (!WT_TIME_WINDOW_HAS_STOP(&vpack->tw) && prepared)) {
```

Comment:
Is this correct? Rollback to stable is supposed to be making decisions based on the durable timestamp, and the HAS_STOP macro is based on the commit timestamp and transaction ID.

Are you also certain that any transaction ID in the time window is cleared after a restart for rollback to stable? i.e: are we appropriately ignoring transaction IDs from previous write generations?

Reply:
This check is to make sure the prepare is on the start time point not on the stop point, so I think it should use the macro. It is possible that we have a zero timestamped stop time point. In that case, the previous check is wrong.

Reply:
Regarding the transaction ids, we call rollback to stable after we set the base write gen in `__recovery_setup_file` so I believe we should have cleared the transaction ids here.